gpu4pyscf/properties/eda.py

```diff
@@ -0,0 +1,1019 @@
+                    mocc_i = mocc_sum[:, nocc_offsets[i_frag] : nocc_offsets[i_frag + 1]]
+                    F_i = Fock_list[i_frag]
+
+                    for j_frag in range(i_frag + 1, n_frag):
+                        mocc_j = mocc_sum[:, nocc_offsets[j_frag] : nocc_offsets[j_frag + 1]]
+                        F_j = Fock_list[j_frag]
+
+                        ij_frag_pair = upper_trinagular_to_pair_index(i_frag, j_frag, n_frag)
+                        x_ij = x[nocc_frag_pair_offsets[ij_frag_pair] : nocc_frag_pair_offsets[ij_frag_pair + 1]]
+                        x_ij = x_ij.reshape(nocc_count[i_frag], nocc_count[j_frag])
+
+                        for k_frag in range(0, n_frag):
+                            mocc_k = mocc_sum[:, nocc_offsets[k_frag] : nocc_offsets[k_frag + 1]]
+                            F_k = Fock_list[k_frag]
+
+                            for l_frag in range(k_frag + 1, n_frag):
+                                mocc_l = mocc_sum[:, nocc_offsets[l_frag] : nocc_offsets[l_frag + 1]]
+                                F_l = Fock_list[l_frag]
+
+                                kl_frag_pair = upper_trinagular_to_pair_index(k_frag, l_frag, n_frag)
+                                y_kl = cp.zeros([nocc_count[k_frag], nocc_count[l_frag]])
+
+                                if k_frag == j_frag:
+                                    orbital_hessian_il = mocc_i.T @ (F_l - 2 * F_k + F_i) @ mocc_l
+                                    y_kl += (orbital_hessian_il.T @ x_ij).T
+                                    orbital_hessian_il = None
+                                if l_frag == j_frag:
+                                    orbital_hessian_ik = - mocc_i.T @ (F_k - 2 * F_l + F_i) @ mocc_k
+                                    y_kl += orbital_hessian_ik.T @ x_ij
+                                    orbital_hessian_ik = None
+                                if l_frag == i_frag:
+                                    orbital_hessian_jk = mocc_j.T @ (F_k - 2 * F_l + F_j) @ mocc_k
+                                    y_kl += (x_ij @ orbital_hessian_jk).T
+                                    orbital_hessian_jk = None
+                                if k_frag == i_frag:
+                                    orbital_hessian_jl = - mocc_j.T @ (F_l - 2 * F_k + F_j) @ mocc_l
+                                    y_kl += x_ij @ orbital_hessian_jl
+                                    orbital_hessian_jl = None
+
+                                y[nocc_frag_pair_offsets[kl_frag_pair] : nocc_frag_pair_offsets[kl_frag_pair + 1]] += \
+                                    y_kl.reshape(nocc_count[k_frag] * nocc_count[l_frag])
+                                y_kl = None
+
+                        x_ij = None
+
+                F_i = None
+                F_j = None
+                F_k = None
+                F_l = None
+
+                return y
+
+            def conjugate_gradient(A_operator, b, x0, tol, maxiter = None):
```

Comment:
* Move the closure out of the "    for cycle in range(mf_sum.max_cycle):" loop. Consider to define it as an independent function.
* Why not using scipy.optimize.minimize(method='CG') ?